packages/flutter/test/cupertino/tab_scaffold_test.dart

```diff
@@ -72,7 +73,7 @@ void main() {
         ),
       ),
     );
-
+    addTearDown(memoryImage.evict);
```

Comment:
It seems we already have `tearDown(memoryImage.evict);` on line 50 for every test.
Is additional `addTearDown(memoryImage.evict);` in every test needed?

Reply:
Interesting. Did you try to add `tearDownAll(imageCache.clear)`?

What you are doing in this PR will become common practice. So, I want to make it right.
And I want to make sure this PR does not hide bugs in 'evict' and `ImageCache`.

Reply:
it seems like i might have not used random seed after refactoring which is why all test case passed for `addTearDown(memoryImage.evict);` and with random seed its not passing unless i use `memoryImage.evict()`
inside test cases

Reply:
> Interesting. Did you try to add `tearDownAll(imageCache.clear)`

this is nice idea i'll try this out

Reply:
And I will involve someone who is more familiar with image caching in Flutter.

Reply:
What exceptions do you get with tearDown? If it is about double disposal, it may mean some tests dispose image and some do not. And fix mat be different.

Reply:
Yes i was disposal errors

`it may mean some tests dispose image and some do not. And fix mat be different.` and yes this turns-out to be true i was able to fix all issue i encountered with some changes except for the `test/widgets/obscured_animated_image_test.dart ` which doesn't pass unless i use `evict` without any teardown functions

can check PR if everything looks fine? i'll try to fix `obscured_animated_image_test.dart` testcase

`And I will involve someone who is more familiar with image caching in Flutter.` That would be nice

Also i found some weird scenario in which if `test/widgets/obscured_animated_image_test.dart ` test does not pass if seed value is `20240539` but passes with other random seeds i'll attach video for reference, is it perhaps just some kind of test-case caching?

https://github.com/flutter/flutter/assets/102401667/0169d655-3862-44ce-93a2-799d59df78e0

Reply:
It turned out that at the moment there is no people around who know image caching in Flutter in details. :(
So, we will need to figure it out.

And the solution may be combination of fixes in the tests and fixes in Flutter code.

I added some documentation on how to work with leaking images: https://github.com/dart-lang/leak_tracker/pull/230 
And, we may need to add more, when we figure out how to fix https://github.com/flutter/flutter/issues/145599

If this feels too complicated, feel free to leave this PR and switch to easier leaks. For example https://github.com/flutter/flutter/issues/145600

Reply:
Tests should pass with all randomization seeds. 
So, that's great that you found seed that fails the test. And that's great that all other tests seem to be fixed.

Can you:
1. Replace 'Fixes ...' with 'Contributes to ...' in the header of the test, so that issue does not close when this PR gets merged.
2. Leave image_icon_test.dart opted out from leak tracking
3. Add the found randomization seed to description of the issue, so that those who will work on it know how to repro.

Reply:
Also, I would like to make sure we are not hiding a real bug.
In real application, which object disposes image provided to ImageIcon, provided to TabBar?
Is disposal happening at all?

Could you put together an application that reproduces this case and check if image is disposed and, if yes, by which class?